packages/tpetra/core/src/Tpetra_Details_PackTraits.hpp

```diff
@@ -249,12 +275,18 @@ struct PackTraits {
       // entries of the input array have the correct size.
       const T& val = packValueCount (outBuf(0));
```

Comment:
@mhoemmen, in another comment you said

> I think I just realized what's going on with PackTraits

was this referring to this part of the code?  I'm still scratching me head over why this works.